db/db_impl/db_impl_write.cc

```diff
@@ -437,10 +437,10 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
         valid_batches += writer->batch_cnt;
         if (writer->ShouldWriteToMemtable()) {
           total_count += WriteBatchInternal::Count(writer->batch);
+          total_byte_size = WriteBatchInternal::AppendedByteSize(
```

Comment:
We have a similar double counting issue in `PipelinedWriteImpl` too. Would you mind fixing that together and add a test too? Thanks!

Reply:
Got it, I'll add that too!

Reply:
I've fixed this and added a test as well